Python/pathconfig.c

```diff
@@ -67,7 +67,7 @@ Py_GetSepA(const char *name)
 #if defined(__MINGW32__)
     msystem = Py_GETENV("MSYSTEM");
 #endif
-    if (msystem != NULL)
+    if (msystem != NULL && strcmp(msystem, "") != 0)
```

Comment:
This may not be safe on non-`_WIN32`, which seems to have been a concern according to the above comment: it sets msystem to `(char*)2` so it is non-NULL on Windows, but now the pointer is actually dereferenced by this `strcmp`.

Reply:
I see. Maybe we can guard the entire block so it only exists on MinGW? I'll push a fix.

Reply:
I've done https://github.com/msys2-contrib/cpython-mingw/commit/2b194e31c3c4741655c33435b13c4182d3c52b9e. I'll squash it into original commit in a bit.